### ISSUE TITLE
Tutorial: instructions for installation prior to the day

### ISSUE THREAD

**Issue by anshumanmohan** [cucapra/calyx#1551]
I don't think we would annoy anyone with a tight list of instructions emailed a week out. It could have a "get as far through this list as you can, but no worries if you get stuck" flavor.

- Install Docker
- Pull our image
- Build it
- Check that you can `run -it` it

**Comment by rachitnigam** [comment]
Good idea! Not sure if we can get registrant's emails before the event but worth asking in the discord

**Comment by sampsyo** [comment]
Maybe a good place to start is with another Markdown document of our own, on the tutorial site, and then if we ever collect attendee emails, we can send a link to that? (We can at least email the virtual attendees; I can extract a list of emails from my side.)

**Comment by anshumanmohan** [comment]
I sorta began to write this out, but then realized that if we do the Docker up correctly, we will need them to literally do what we say [here](https://docs.calyxir.org/#using-the-docker-container), i.e.
```
docker pull ghcr.io/cucapra/calyx:latest
docker run -it --rm ghcr.io/cucapra/calyx:latest
```

The first step will take some time, and possibly some troubleshooting that they can do themselves (e.g. is my Docker daemon running? Ans: [here](https://stackoverflow.com/a/44719239))

If we want to be extra careful, we could ask them to run two more things inside the Docker:
```
fud check
runt -i 'core | mrxl'
```
(the `fud check` may get tighter so that it is 100% successful; see #1559)

but ideally we should _know with such certainty_ that those two checks will work that we don't even need to ask them to run it, haha

**Comment by rachitnigam** [comment]
> but ideally we should know with such certainty that those two checks will work that we don't even need to ask them to run it, haha

Physicists have long searched for a perfectly spherical world with uniform gravity; CS people have searched for a reproducible build system. Both communities have not yet found them...

(Jokes aside: Docker is a good start and yes, it would be nice to have that guarantee but I don't trust docker enough to make that claim)

**Comment by anshumanmohan** [comment]
Cool cool, #1556 should merge shortly and that asks them to run `runt -i core`. 

If you give me the short and sweet incantation `fud` incantation that won't scare anyone, I'll add that too. That will close #1559.

Then we'll be able to close this issue with an email to people saying, "Please go [here](https://docs.calyxir.org/#using-the-docker-container) and paste 2+2 lines of code into your machine."

**Comment by rachitnigam** [comment]
This is the format of the `fud check` command (just a list of stages we expect to be installed correctly): https://github.com/cucapra/filament-evaluation/blob/main/scripts/configure-fud.sh#L53

Can you add the stages that you think should be in that list? Probably `verilator`, `jq`, `mrxl`, etc.

**Comment by rachitnigam** [comment]
@anshumanmohan what's the actionable thing to do here? Do we try to get the emails of interested people and send them the information on Saturday?

**Comment by anshumanmohan** [comment]
Yes, specifically we point them to the [Docker installation instructions](https://docs.calyxir.org/#using-the-docker-container) and tell them to follow that little section. It's four commands, two copy-pastes.

Heads up that the docs don't seem to have the latest docker tag. That's required before we send this email.

**Comment by anshumanmohan** [comment]
Or rather, they have `latest` and not the one we want them to use.

**Comment by anshumanmohan** [comment]
I kinda think it should go out ASAP, not on Saturday, but not a big deal

**Comment by rachitnigam** [comment]
Okay, cool! Once the change is merged, we can send it out? Have you asked Steve for a list of people already? Also, let's make the sign off "The Calyx Team" instead of "Rachit and the Gang"

**Comment by anshumanmohan** [comment]
I [asked](https://github.com/cucapra/calyx/issues/1551#issuecomment-1581072499) on the Discord once but we'll need to prod Steve again tomorrow. If we're emailing/PMing someone I wonder if Steve is the right person. Might Zach/Nadia be better, seeing as they are chairs of tutorials and workshops?

**Comment by rachitnigam** [comment]
So looks like there is a channel for our workshop. Maybe we can also post the materials and the email text there?

**Comment by anshumanmohan** [comment]
I had a thought. What if folks think that we're asking them to run the image at home and then keep it alive all way until the workshop? The copy in the email needs something like "you can then close the image with `exit`. `docker run` will automatically go faster the next time you run it."

Relatedly, taking @sampsyo's tip: maybe we can simplify the instruction (in the docs) by plain removing the `docker pull` step? It happens automatically, I hear?

Finally, one more small thing I'd like to the docs: before the second code block, I'd like to say "now working within the Docker image, run...`

**Comment by sampsyo** [comment]
Sounds good! I hope we don't have to be _too_ hand-holdy w/r/t generic Docker instructions (e.g., how to leave a `docker run -it` session), but I suppose it can't hurt.

**Comment by anshumanmohan** [comment]
I've sent [this](https://github.com/cucapra/calyx/issues/1551#issuecomment-1589806496) to our Discord channel. It would be great to send it to the remote attendees too. I lack the list of emails. Just a heads up @sampsyo! Feel free to either send me the list or to send out the text, whichever is easier!
